### DISCUSSION TITLE
[Feature Request] Standalone clipboard component

### DISCUSSION THREAD

**Discussion by timelf123** [heroui-inc/heroui#4686] [Feature requests]
### Is your feature request related to a problem? Please describe.

`Snippet` is the only component with clipboard functionality. I want a drop in `ClipboardButton` component to be used in the startContent/endContent of other components like `Input` etc

### Describe the solution you'd like

As a user I expected to see reusable clipboard functionality with matched animations etc.

Steps to fix:
rework clipboard portion of `Snippet` into shareable component and use it in `Snippet` component, 

### Describe alternatives you've considered

Disabling clipboard on snippets and building my own - doable but seems silly

**Comment by TheAdail** [comment]
As another use case, I could not find how to copy to clipboard from the Table component, so maybe this Clipboard component could be used for that as well.

**Comment by henry-oway** [comment]
+1
